server/plugin/api_test.go

```diff
@@ -62,6 +62,144 @@ func TestWithRecovery(t *testing.T) {
 	}
 }
 
+func TestPlugin_handleSkip(t *testing.T) {
+	httpTestJSON := testutils.HTTPTest{
+		T:       t,
+		Encoder: testutils.EncodeJSON,
+	}
+
+	for name, test := range map[string]struct {
+		httpTest          testutils.HTTPTest
+		request           testutils.Request
+		expectedResponse  testutils.ExpectedResponse
+		ParseAuthTokenErr error
+		LoadUserErr       error
+		callError         error
+	}{
+		"Skip message is successfully sent to virtual Agent": {
+			httpTest: httpTestJSON,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("%s%s", pathPrefix, PathSkip),
+				Body: model.PostActionIntegrationRequest{
+					ChannelId: "mock-channelID",
+				},
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+		},
+		"Error while decoding response body": {
+			httpTest: httpTestJSON,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("%s%s", pathPrefix, PathSkip),
+				Body:   nil,
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
```

Comment:
Why is it `StatusOK` in case of error?

Reply:
check everywhere

Reply:
for now in this case we have `StatusCode` is `Unauthorized` when user is unauthorized so added that.

Reply:
Can you explain this in detail?

Reply:
I have done the `StatusCode` as `Unauthorized` when user is not autherised please see https://github.com/Brightscout/mattermost-plugin-servicenow-virtual-agent/blob/5e2e79ce2473178428b02d948d8024cb6404d0f2/server/plugin/api_test.go#L95